routers/repo/issue_dependency.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2018 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"fmt"
+	"net/http"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/context"
+)
+
+// AddDependency adds new dependencies
+func AddDependency(ctx *context.Context) {
+
+	depID := ctx.QueryInt64("newDependency")
+
+	issueIndex := ctx.ParamsInt64("index")
+	issue, err := models.GetIssueByIndex(ctx.Repo.Repository.ID, issueIndex)
+	if err != nil {
+		ctx.ServerError("GetIssueByIndex", err)
+		return
+	}
+
+	// Check if the Repo is allowed to have dependencies
+	if !ctx.Repo.CanCreateIssueDependencies(ctx.User) {
+		ctx.NotFound("NotAllowedToCreateIssueDependencies", nil)
+		return
+	}
+
+	// Dependency
+	dep, err := models.GetIssueByID(depID)
+	if err != nil {
+		ctx.Flash.Error(ctx.Tr("repo.issues.dependency.add_error_dep_not_exist"))
+		url := fmt.Sprintf("%s/issues/%d", ctx.Repo.RepoLink, issueIndex)
```

Comment:
You can probably add before everything a `defer ctx.Redirect(fmt.Sprintf("%s/issues/%d", ctx.Repo.RepoLink, issueIndex), http.StatusSeeOther)` without having to copypaste the code every time.